bokeh/models/annotations.py

```diff
@@ -1048,6 +1048,10 @@ class Tooltip(Annotation):
 
     inner_only = Bool(default=True, help="""
     Whether to display outside a central plot frame area.
+
+    .. note:
+        This property is deprecated and will be removed in bokeh 3.0.
```

Comment:
Can you explain this more? Normally we don't want hover tooltips to show unless the cursor is inside the plot frame. Are you just making that the unconditional behavior all the time?

Reply:
Also just to reiterate: I would like to deprecate the entire `ToolTip` model. It has a bad name, and importantly , it's never actually been usable by users, so this should not be a stretch.

Reply:
Now `HoverTool` decides whether a tooltip should be visible or not, decoupling it from the frame (for the most part).